provider/datagen/src/transform/icuexport/collator/test.rs

```diff
@@ -0,0 +1,181 @@
+            expectation: Ordering::Less,
+        },
+        // ICU4C has a duplicate of the case below.
+        // The duplicate is omitted here.
+        // Instead, the subsequent tests are added for ICU4X.
+        TestCase {
+            left: "ä",
+            right: "o",
+            expectation: Ordering::Greater,
+        },
+        TestCase {
+            left: "a\u{0308}",
+            right: "ä",
+            expectation: Ordering::Equal,
+        },
+    ];
+    let locale: Locale = langid!("fi").into();
+
+    let any_dyn_provider = get_provider();
+    let any_provider = any_dyn_provider.as_any_provider();
+    let provider = any_provider.as_downcasting();
+
+    let mut options = CollatorOptions::new();
+    options.set_strength(Some(Strength::Tertiary));
+
+    {
+        let collator: Collator = Collator::try_new(locale.clone(), &provider, options).unwrap();
+        check_expectations(&collator, &cases);
+    }
+
+    options.set_strength(Some(Strength::Primary));
+
+    {
+        let collator: Collator = Collator::try_new(locale, &provider, options).unwrap();
+        check_expectations(&collator, &cases);
+    }
+}
+
+#[test]
+fn test_sv() {
+    // This is the same as test_fi. The purpose of this copy is to test that
+    // Swedish defaults to "reformed", which behaves like Finnish "standard",
+    // and not to "standard", which behaves like Finnish "traditional".
+
+    // Adapted from ficoll.cpp in ICU4C
+    // Testing that w and v behave as in the root collation is for checking
+    // that the sorting collation doesn't exhibit the behavior of the search
+    // collation, which (somewhat questionably) treats w and v as primary-equal.
+    let cases = [
+        TestCase {
+            left: "wat",
+            right: "vat",
+            expectation: Ordering::Greater,
+        },
+        TestCase {
+            left: "vat",
+            right: "way",
+            expectation: Ordering::Less,
+        },
+        TestCase {
+            left: "aübeck",
+            right: "axbeck",
+            expectation: Ordering::Greater,
+        },
+        TestCase {
+            left: "Låvi",
+            right: "Läwe",
+            expectation: Ordering::Less,
+        },
+        // ICU4C has a duplicate of the case below.
+        // The duplicate is omitted here.
+        // Instead, the subsequent tests are added for ICU4X.
+        TestCase {
+            left: "ä",
+            right: "o",
+            expectation: Ordering::Greater,
+        },
+        TestCase {
+            left: "a\u{0308}",
+            right: "ä",
+            expectation: Ordering::Equal,
+        },
+    ];
+    let locale: Locale = langid!("sv").into();
+
+    let any_dyn_provider = get_provider();
+    let any_provider = any_dyn_provider.as_any_provider();
+    let provider = any_provider.as_downcasting();
+
+    let mut options = CollatorOptions::new();
+    options.set_strength(Some(Strength::Tertiary));
+
+    {
```

Comment:
Mit: the scopes aren't necessary, you could even inline the Collator construction into the check